### ISSUE TITLE
How to build it in ubuntu16.04?

### ISSUE THREAD

**Issue by lsongseven** [NifTK/apriltags#1]
Could you please specify how to build it in ubuntu16.04 or explain the steps?  Googled a lot but still cannot build it.

**Comment by MattClarkson** [comment]
Did you try cmake?

**Comment by lsongseven** [comment]
Oh thanks, finally make it. I thought there were something related with
pods according to readme, but just find that cmake is enough.

**Comment by jackm** [comment]
Just to clarify, is this the correct way to build the package?

Starting at the repository root directory:
```
cd cmake
cmake ..
```

Do I need to run make as well? When I try to do that, I get the following error:
```
Scanning dependencies of target apriltags
[  6%] Building CXX object CMakeFiles/apriltags.dir/src/FloatImage.cc.o
[ 13%] Building CXX object CMakeFiles/apriltags.dir/src/Quad.cc.o
/home/user/apriltags/src/Quad.cc:1:23: fatal error: Eigen/Dense: No such file or directory
compilation terminated.
CMakeFiles/apriltags.dir/build.make:86: recipe for target 'CMakeFiles/apriltags.dir/src/Quad.cc.o' failed
make[2]: *** [CMakeFiles/apriltags.dir/src/Quad.cc.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/apriltags.dir/all' failed
make[1]: *** [CMakeFiles/apriltags.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2
```

I get the same error if I try to run make from the repo root directory as well.

**Comment by MattClarkson** [comment]
No. You should not build within the source folder. Given you have
```
/home/user/apriltags/src/
```

I suggest you delete it and reclone the src repository to make sure it is clean. Then,

```
mkdir /home/user/apriltags/build
cd /home/user/apriltags/build
ccmake ../src
```
set whichever options you need, but probably leave default. Then generate.
Then compile with 
```
make
```
If you are unsure of CMake try stuff like: https://cliutils.gitlab.io/modern-cmake/chapters/intro/running.html

**Comment by jackm** [comment]
Still no good. I believe that cmake is not finding or was not given the eigen3 library include directory.  On my system running Ubuntu 16.04.5 LTS, the eigen library is found at `/usr/include/eigen3/`.

A quick hack was to add the line `set(Eigen_DIR "/usr/include/eigen3/")` to the root CMakeLists.txt before the first `configure_file()` line. After this, I was able to compile everything using make without errors.

Could you please update the main README to include build steps?

**Comment by onlineth** [comment]
This should definitely be added, thank you @jackm
